src/schema/sale/display.ts

```diff
@@ -0,0 +1,98 @@
+import moment from "moment"
+
+export const isLiveOpen = sale => {
+  const liveStart = moment(sale.live_start_at)
+  return (
+    sale.auction_state === "open" &&
+    (moment().isAfter(liveStart) || moment().isSame(liveStart))
+  )
+}
+
+const hasEnded = end_at => {
+  return end_at && end_at < moment()
+}
+
+export async function displayTimelyAt(
+  sale,
+  useUpdatedFormatting: boolean,
+  meBiddersLoader
+) {
+  const { live_start_at, registration_ends_at, start_at, end_at } = sale
+
+  // Don't display any info for past auctions.
+  if (useUpdatedFormatting && hasEnded(end_at)) {
+    return null
+  }
+
+  // Display 'ended' for past auctions in legacy contexts.
+  if (hasEnded(end_at)) {
+    return "Ended"
+  }
+
+  // Sale requires registration.
+  // Display 'register by' label if there is an unregistered bidder
+  // and the registration period is open.
+  if (registration_ends_at > moment()) {
+    let isRegistered = false
+
+    // Check if there is a logged in user which is registered.
+    if (meBiddersLoader) {
+      const bidders = await meBiddersLoader({ sale_id: sale.id })
+      isRegistered = bidders && bidders.length > 0
+    }
+    if (!isRegistered) {
+      const diff = moment().diff(moment(registration_ends_at), "hours")
+      const format = diff > -24 ? "ha" : "MMM D, ha"
+      const label = `Register by\n${moment(registration_ends_at).format(
+        format
+      )}`
+      return label
+    }
+  }
+
+  // Live Auction (already registered or registration closed).
+  if (live_start_at) {
+    // Auction is currently in progress.
+    if (isLiveOpen(sale)) {
+      return "In Progress"
+    }
+
+    // Auction hasn't started yet.
+    if (live_start_at) {
+      return `Live ${moment(live_start_at).fromNow()}`
+    }
+  }
+
+  // Timed Auction
+  if (start_at) {
+    const range = moment().add(5, "days")
+    const startAt = moment(start_at)
+    const isInProgress = startAt < moment()
+    const isUpcoming = startAt > moment() && startAt < range
+    const isNearFuture = startAt > range
+    const dateLabel = saleDate => {
```

Comment:
This is another place where I might branch based on the `useUpdatedFormatting` flag.

The current setup here is to display things like 'X days/min left' , whereas the new spec has 'ends in X days/min'.